sdk/core/azure-core/src/main/java/com/azure/android/core/http/options/RequestOptions.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.core.http.options;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.azure.android.core.http.ServiceClient;
+import com.azure.android.core.util.CancellationToken;
+
+/**
+ * Options for a service operation to be carried out by a {@link ServiceClient}.
+ */
+public class RequestOptions {
+    @Nullable
+    String clientRequestId;
```

Comment:
Do we want to expose an option for the user to set the RequestId? In Java SDK, the generation and setting of the request-id header is hidden in the default RequestId policy.